multinode_integration_tests/src/masq_node.rs

```diff
@@ -26,20 +30,30 @@ use std::time::Instant;
 pub struct NodeReference {
     pub public_key: PublicKey,
     pub node_addr_opt: Option<NodeAddr>,
+    pub chain_id: u8,
 }
 
 impl FromStr for NodeReference {
     type Err = String;
 
+    //TODO inaccurate - this works only as long as we don't parse the real Node descriptor including its URL prefix which we don't now, see extract_node_reference()
```

Comment:
With some slight changes this could almost be implemented using standard URL/URI classes where:
The standard URI is:
<protocol>://<user>:<password>@<host or IP>:<port>
So instead of
masq://eth.QUJDREVGR0hJSktMTU5PUFFSU1RVVldYWVowMTIzNDU:1.2.3.4:1234
we have
masq://eth:QUJDREVGR0hJSktMTU5PUFFSU1RVVldYWVowMTIzNDU@1.2.3.4:1234
protocol = masq
user = eth
password = QUJDREVGR0hJSktMTU5PUFFSU1RVVldYWVowMTIzNDU
host/ip = 1.2.3.4
port = 1234

`user` and `password` aren't actually user and password but the values we want to use in their place.